src/sentry/testutils/helpers/eventprocessing.py

```diff
@@ -14,6 +16,18 @@ def write_event_to_cache(event):
     return event_processing_store.store(cache_data)
 
 
+# Wrap the real `Event` type in order to let mypy know that certain attributes are guaranteed
+# to be defined
+class Event(_Event):
+    group: Group
+    group_id: int
+
+    # Note: This function never gets called , but writing it out this way is necessary to change the
+    # return type in mypy's eyes
+    def get_primary_hash(self) -> str:
+        return NonNone(super().get_primary_hash())
```

Comment:
Yes, you're right, there's still one in there. And I can split it out into `primary_hash = ...; assert primary_hash; return primary_hash` if I must but that's exactly (and all) the `NonNone` is doing. I guess I really don't see how a helper whose only job it to assert something exists is so problematic.

Reply:
sometimes when you set up types you're going to have to fix types in other modules -- that's ok!  often individuals of the past annotated things ~slightly wrong because our typing system wasn't set up properly.  sometimes you're going to have to put in a tiny bit of extra work to make everything better for everyone.

this is one of those examples!  the function you've been hacking around is probably typed incorrectly and with a little bit of investigation (a metric if it's `None`, probably never fired) you should be able to fix that and then make everything better for everyone.

my concern with this NonNone thing is that it makes it extremely easy to unsafely cast away nullability.  where the correct fix is usually to handle it as an error *or* fix the underlying nullability at its source. that is why this "helper" should have never existed in the first place and why I'm so strongly trying to discourage its use now

I understand that this probably wasn't obvious but that's why I'm here now. you can always add me or the typing group to your typing PRs if you want earlier feedback.

also, here are some  [guidelines](https://www.notion.so/sentry/sentry-typing-guidelines-best-practices-471d5aacf262476188982ce70f909e00) we have created and a reminder about [cast](https://www.notion.so/sentry/sentry-typing-guidelines-best-practices-471d5aacf262476188982ce70f909e00?pvs=4#4c5f66d5e5f2460991cf1074111fdd85) as well as `# type: ignore` / etc from our best practices document